easybuild/easyconfigs/b/BamTools/BamTools-2.5.0-foss-2016b.eb

```diff
@@ -0,0 +1,28 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+
+easyblock = 'CMakeMake'
```

Comment:
@SethosII There's a dedicated easyblock for `BamTools`, that should be updated for recent `BamTools` versions instead of using the generic `CMakeMake` easyblock?

Reply:
@boegel I think a dedicated easyblock isn't needed anymore as of [version 2.4.2](https://github.com/pezmaster31/bamtools/releases/tag/v2.4.2):

> CMake has been sanitized and made more modern, idiomatic and standardized. Out-of-source builds are guaranteed now. Instead of picking binaries/libraries from the build tree, users should stage and/or install BamTools instead.

Whats do you think?

Reply:
@SethosII When we have an existing software-specific easyblock, we try to keep it updated and used in all easyconfig for that software, since it allows trimming the easyconfigs quite a bit (e.g. `sanity_check_paths`, etc.).

Are you up for updating the `BamTools` easyblock? If not, can you let me know what would need to change?

Reply:
@boegel Sorry, my understanding of easyblocks is to limited right now so I won't be able to figure it out on my own. The changes needed from 2.4.2 (based on the release notes) onwards are these:

* `MakeCp` part is superflous, normal install will work
* only the static library is build, so no *.so files anymore (you can pass `-DBUILD_SHARED_LIBS=ON` but it will only build shared object files then, no direct way for building static and shared)
* the headers have an additional subdirectory `bamtools`
* make install doesn't copy docs

So the `custom_paths` variable should be altered like this:

```
custom_paths['files'].remove("lib/libbamtools.%s" % sharedlib_ext)
custom_paths['files'].remove("include/shared/bamtools_global.h")
custom_paths['files'].append("include/bamtools/shared/bamtools_global.h")
custom_paths['dirs'].remove("include/api")
custom_paths['dirs'].append("include/bamtools/api")
custom_paths['dirs'].remove("docs")
```

Reply:
@SethosII Updated `BamTools` easyblock is available in https://github.com/easybuilders/easybuild-easyblocks/pull/1332, PR for BamTools 2.5.0 (with `intel/2017b`) in #5576